tools/yul_phaser/Chromosome.h

```diff
@@ -0,0 +1,55 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+
+namespace solidity::phaser
+{
+
+class Chromosome
+{
+public:
+	Chromosome():
+		m_optimizationSteps{} {}
+	explicit Chromosome(std::vector<std::string> const& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	explicit Chromosome(std::vector<std::string>&& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	static Chromosome make_random(int _length);
```

Comment:
```suggestion
	static Chromosome makeRandom(int _length);
```

Reply:
Also, as I assume `_length` should not be negative, please use `size_t`.

Reply:
> Also, as I assume `_length` should not be negative, please use `size_t`.

I'm actually a bit conflicted about this. On one hand the values I depend on internally are usually `size_t` so it makes sense to use it too. On the other hand the [guidelines say to prefer signed types for this](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#arithmetic). I eventually settled on just sticking to `size_t` but looks like I missed a few places when I was changing that. What is your stance on that? Stick to the guidelines or just don't worry and use unsigned types for non-negative values?

Reply:
I actually was not aware that subscripts use signed integers. My point here, and I think this overrules the other, is that this is about the external interface of this class. If it does not make sense to call this with a negative number, then the type system should already reject it. Same rule about using pointer or reference: Use pointer only if nullptr is a valid value.

Now what the function does internally is another thing. It could cast to signed (and do length check) before doing index access, but that would probably be overkill.

So to summarize: I would use `size_t` to avoid confusion about the semantics of this function when not looking at any documentation or implementation.

Reply:
ok. Done.